com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DeleteServicesStep.java

```diff
@@ -54,6 +54,17 @@ protected StepPhase executeAsyncStep(ExecutionWrapper execution) throws Exceptio
 
             List<String> servicesToDelete = StepsUtil.getServicesToDelete(execution.getContext());
 
+            if (servicesToDelete.isEmpty()) {
+                getStepLogger().debug(Messages.MISSING_SERVICES_TO_DELETE);
+                return StepPhase.DONE;
+            }
+
+            if (!StepsUtil.getDeleteServicesFlag(execution)) {
+                getStepLogger().warn(Messages.SKIP_SERVICES_DELETION, String.join("\", \"", servicesToDelete));

```

Comment:
Final decision: Do not log the services names as part of the message, because the message may become too long as a result. We considered logging them only on "debug", so that the user will not see the long message, but even that is unnecessary because they are already logged in BuildCloudUndeployModelStep.

Sorry for all the back and forth discussions and changing requirements.